### PR TITLE
refactor(hooks): provide input ref and fix setFiles issue on Firefox

### PR DESCRIPTION
**Pull request by neighborhood999**
1. `useEffect` on [Firefox issue](https://codesandbox.io/s/input-file-with-useeffect-demo-yrmgk)

2. I think hook can accept incoming `ref` or provide by self, e.g:

```ts
const { ref, files } = useInputFile<HTMLInputElement>();
//
// awsome UI
// ...
// <input ref={ref} />
// ...
```

or

```ts
const userCustomRef = useRef<HTMLInputElement>(null);
const { files } = useInputFile({ ref: userCustomRef });
```

3. Moving onChange callback as an optional argument, user can custom their `onChange`:

```ts
const onChange = (event: Event) => {
  console.log(event.currentTarge);
  // ...
};
const { files } = useInputRef({ onChange });
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  const ref = args.ref || defaultRef;
+  customHandler.current = args.onChange;
+
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.setAttribute('type', 'file');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (bluebill1049) [inline src/index.ts]: maybe leave `ref` in there? it's not going to be picked up by the diff at least without the eslint disable.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+function useInputFile(
+  args: {
+    ref?: RefObject<HTMLInputElement>;
+    options?: Options | undefined;
+    onChange?: OnChange;
+  } = {}
+): Return {
+  const [files, setFiles] = useState<File[] | null>(null);
+  const defaultRef = useRef<HTMLInputElement>(null);
```
Comment (bluebill1049) [inline src/index.ts:25]: why do you need this?

Comment (neighborhood999) [reply src/index.ts:25]: If user didn't provide `ref`, I want to provide ref by hook.

File: src/index.ts (hunk 1 above)
Comment (neighborhood999) [reply src/index.ts]: OK 👍

**Pull request merged**